code/weapon/weapons.cpp

```diff
@@ -3380,6 +3390,29 @@ void weapon_generate_indexes_for_substitution() {
 
 			memset(wip->weapon_substitution_pattern_names, 0, sizeof(char) * MAX_SUBSTITUTION_PATTERNS * NAME_LENGTH);
 		}
+
+		if (wip->failure_rate > 0.0f) {
+			if (stricmp("none", wip->failure_sub_name) != 0) {
```

Comment:
You could use `VALID_FNAME` here to ensure consistent behavior with other parts of the code.